mlem/cli/types.py

```diff
@@ -1,59 +1,63 @@
-        if module != "builtins" and "." not in type_name:
-            type_name = f"{module}.{type_name}"
-        type_name = color(type_name, "yellow")
-
-        if field.required and not force_not_req:
-            req = color("[required] ", "grey")
-        else:
-            req = color("[not required] ", "white")
-        if not field.required:
-            default = field.default
-            if isinstance(default, str):
-                default = f'"{default}"'
-            default = f" = {default}"
-        else:
-            default = ""
-        if (
-            isinstance(field.type_, type)
-            and issubclass(field.type_, MlemABC)
-            and field.type_.__is_root__
-        ):
-            echo(
-                req
-                + color(fullname, "green")
-                + ": One of "
-                + color(f"mlem types {field.type_.abs_name}", "yellow")
+def _add_examples(generator: Iterator[CliTypeField], parent_help=None):
+    for field in generator:
+        field.help = parent_help or field.help
+        yield field
+        if field.is_list or field.is_mapping:
+            key = ".key" if field.is_mapping else ".0"
+            yield from _add_examples(
+                parse_type_field(
+                    path=field.path + key,
+                    type_=field.type_,
+                    help_=field.help,
+                    is_list=False,
+                    is_mapping=False,
+                    required=False,
+                    allow_none=False,
+                    default=None,
+                ),
+                parent_help=f"Element of {field.path}",
             )
-        elif isinstance(field.type_, type) and issubclass(
-            field.type_, BaseModel
-        ):
-            echo(req + color(fullname, "green") + ": " + type_name)
-            explain_type(field.type_, fullname, not field.required)
-        else:
-            echo(req + color(fullname, "green") + ": " + type_name + default)
+
+
+def type_fields_with_collection_examples(cls):
+    yield from _add_examples(iterate_type_fields(cls))
+
+
+def explain_type(cls: Type[BaseModel]):
+    echo(
+        color("Type ", "")
+        + color(cls.__module__ + ".", "yellow")
+        + color(cls.__name__, "green")
+    )
+    if issubclass(cls, MlemABC):
+        echo(color("MlemABC parent type: ", "") + color(cls.abs_name, "green"))
+        echo(color("MlemABC type: ", "") + color(cls.__get_alias__(), "green"))
+    if issubclass(cls, MlemObject):
+        echo(
+            color("MlemObject type name: ", "")
+            + color(cls.object_type, "green")
+        )
+    echo((cls.__doc__ or "Class docstring missing").strip())
+    fields = list(type_fields_with_collection_examples(cls))
+    if not fields:
+        echo("No fields")
+    else:
+        echo("Fields:")
+    for field in fields:
+        echo(field.to_text())
 
 
 @mlem_command("types", hidden=True)
```

Comment:
for model, probably should use it like `mlem types meta model`, so model is not valid since it's not root class. probably the error is the same as second one, so we should just check if `x` is valid root/implementation name and print error with a list of valid values if it is not

Reply:
I just got empty output from `mlem types model`. But still fixed both

Reply:
Checked, it works. Thanks!